### PR TITLE
feat(ProspectPage): added prospect page

### PR DESCRIPTION
**Pull request by FastNinja**

### CODE REVIEW

File: client/src/prospects/ProspectDetails.jsx [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+  }
+}
+
+ProspectDetails.propTypes = {
+  prospect: PropTypes.object.isRequired,
+};
+
+export default compose(
+  withStyles(styles),
+  withApiClient
```
Comment (dejanvasic85) [inline client/src/prospects/ProspectDetails.jsx]: don't need this

File: client/src/prospects/ProspectDetailsPage.jsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+});
+
+class ProspectDetailsPage extends React.Component {
+  state = {
+    isFetching: true,
+    prospect: {},
+  };
+
+  async componentDidMount() {
+    const prospectId = this.props.match.params.id;
```
Comment (dejanvasic85) [inline client/src/prospects/ProspectDetailsPage.jsx:16]: you can destruct:
```
const { api } = this.props;
```

File: client/src/prospects/ProspectDetailsPage.jsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    const prospect = await this.props.api.getProspect(prospectId);
```
Comment (dejanvasic85) [inline client/src/prospects/ProspectDetailsPage.jsx:17]: same as here

File: client/src/prospects/ProspectDetailsPage.jsx [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    this.setState({
+      prospect,
+      isFetching: false,
+    });
+  }
+
+  handleOnNewGift = clientId => {
+    this.props.history.push(`/app/clients/${clientId}/gifts/new`);
```
Comment (dejanvasic85) [inline client/src/prospects/ProspectDetailsPage.jsx]: don't use history.push please?

File: client/src/prospects/ProspectList.jsx [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+import { withStyles } from '@material-ui/core/styles';
+import { List, Typography, Card, CardContent, CardHeader, Divider } from '@material-ui/core';
+import ProspectListItem from './ProspectListItem';
+
+const styles = theme => ({
+  root: {
+    cursor: 'pointer',
+  },
+  avatar: {
```
Comment (dejanvasic85) [inline client/src/prospects/ProspectList.jsx]: not used.

File: client/src/prospects/ProspectList.jsx [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    backgroundColor: theme.palette.background.paper,
+  },
+  inline: {
+    display: 'inline',
+  },
+});
+
+export class ProspectList extends React.Component {
+  handleProspectClick = prospect => {
+    this.props.history.push(`/app/prospects/${prospect.id}`);
```
Comment (dejanvasic85) [inline client/src/prospects/ProspectList.jsx]: using history.push again. Let's try get rid of all of them.

File: client/src/prospects/ProspectList.jsx [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+      <Card>
+        <CardHeader title="Prospects" />
+        <Divider />
+        <CardContent>
+          <List className={classes.prospects}>
+            {prospects.map(prospect => (
+              <ProspectListItem
+                key={prospect.id}
+                prospect={prospect}
+                onClick={() => this.handleProspectClick(prospect)}
```
Comment (dejanvasic85) [inline client/src/prospects/ProspectList.jsx]: you can get rid of this.

File: client/src/prospects/ProspectListItem.jsx [hunk 8]
```diff
@@ -16,44 +13,44 @@ const styles = theme => ({
...
+    return (
+      <ListItem alignItems="center" className={classes.root} component={Link} to={`/app/prospects/${prospect.id}`}>
+        <PersonAvatar details={prospect} />
+        <ListItemText
+          primary={`${prospect.firstName} ${prospect.lastName}`}
+          secondary={
+            <React.Fragment>
               <Typography component="span" className={classes.inline} color="textPrimary">
-                Referred by 
+                {prospect.email}
```
Comment (dejanvasic85) [inline client/src/prospects/ProspectListItem.jsx:34]: We should work on the designs later but this should stand out a LOT more. And should be a link to the Gift page with prefilled data.

File: client/src/prospects/ProspectListItem.jsx [hunk 9]
```diff
@@ -16,44 +13,44 @@ const styles = theme => ({
...
-          </React.Fragment>
-        }
-      />
-    </ListItem>;
+              &nbsp; {prospect.phone}
+              <div>
+                <Typography component="span" className={classes.inline} color="textPrimary">
+                  Referred by
+                </Typography>
+                &nbsp; {prospect.Client.firstName} {prospect.Client.lastName}
```
Comment (dejanvasic85) [inline client/src/prospects/ProspectListItem.jsx:41]: Why is this capital letter `Client`?

File: server/routes/clients.js [hunk 10]
```diff
@@ -52,11 +52,21 @@ router.get(
       order: [['createdAt', 'DESC']]
     });
 
+    const prospects = await Prospect.findAll({
+      where: { clientId: id, agentId: agentId },
+      include: [{
```
Comment (dejanvasic85) [inline server/routes/clients.js:57]: That's awesome man! Nice theft from github.

**Review by dejanvasic85 — APPROVED** [review]
Good man

File: client/src/prospects/ProspectListItem.jsx (hunk 9 above)
Comment (FastNinja) [reply client/src/prospects/ProspectListItem.jsx:41]: this is how Sequeliser returns joined tables. I think API can convert result to make it small case...

**Pull request merged**